src/main.rs

```diff
@@ -201,9 +178,14 @@ mod app {
     #[task(priority = 1, local=[pwm], shared=[dac, settings, gpio], capacity=1)]
     fn settings_update(mut c: settings_update::Context, settings: Settings) {
         // Verify settings and make them available
-        c.shared
-            .settings
-            .lock(|current_settings| *current_settings = settings);
+        c.shared.settings.lock(|current_settings| {
+            *current_settings = settings;
+            // Limit y_min and y_max values here. Will be incorporated into miniconf response later.
+            current_settings.output_channel.iter_mut().for_each(|ch| {
```

Comment:
I'd modify `settings` (the input). That reduced the lock time of `shared.settings`.

Reply:
You get the input settings as owned (not as a borrow), right? Then adding the `mut` is just semantics.

Reply:
Right.